docs/source/class_basics.rst

```diff
@@ -315,3 +315,34 @@ class, including an abstract method defined in an abstract base class.
 
 You can implement an abstract property using either a normal
 property or an instance variable.
+
+Slots
+*****
+
+When a class has explicitly defined
+`__slots__ <https://docs.python.org/3/reference/datamodel.html#slots>`_
+mypy will check that all assigned attributes are listed there,
+otherwise an error will be raised.
+
+.. code-block:: python
+
+  class Album:
+      __slots__ = ('name', 'year')
+
+      def __init__(self, name: str, year: int) -> None:
+         self.name = name
+         self.year = year
+         self.released = True  # E: Trying to assign name "released" that is not in "__slots__" of type "Album"
+
+  my_album = Album('Songs about Python', 2021)
+
+We have several rules when we count classes as "slotted"
+(the rules are the same as in CPython):
+
+1. All base classes (except builtin ones) must have explicit ``__slots__`` defined
+2. ``__slots__`` must not include ``__dict__``. For example, ``__slots__ = ("__dict__", ...)`` is not valid.
```

Comment:
Pretty sure CPython allows `__slots__` to include `__dict__`.

> If dynamic assignment of new variables is desired, then add '__dict__' to the sequence of strings in the __slots__ declaration.

So this would fall under custom rules

Reply:
It is very debatable whether slotted class with `__dict__` is still a slotted class 🤔 
Probably it can be in any category.

Reply:
Oh, I see where I got confused, suggested some phrasing changes :-)